### PR TITLE
Fix a number of issues with unsigned event objects

### PR DESCRIPTION
**Pull request by turt2live**
The commits hopefully somewhat explain themselves.

* `UnsignedData` now shows up in the spec as an actual thing (for documentation purposes).
* Fixed `m.room.member`'s schema definition to put `unsigned` in the right place (was under `content`), and have it inherit the docs from the rest of the spec.
* Fixed `m.room.server_acl`'s `age` example being in the wrong place by making it inherit from state events.

### CODE REVIEW

File: event-schemas/schema/core-event-schema/unsigned_prop.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      sent. This field is generated by the local homeserver, and may be incorrect
+      if the local time on at least one of the two servers is out of sync, which can
+      cause the age to either be negative or greater than it actually is.
+    type: integer
+  redacted_because:
+    description: Optional. The event that redacted this event, if any.
+    title: Event
+    type: object
+  transaction_id:
+    description: The client-supplied transaction ID, if the client being given
```
Comment (uhoreg) [inline event-schemas/schema/core-event-schema/unsigned_prop.yaml:30]: Maybe something like

```suggestion
    description: The client-supplied transaction ID, for example, provided via ``PUT /_matrix/client/r0/rooms/{roomId}/send/{eventType}/{txnId}``, if the client being given
```

?

Comment (turt2live) [reply event-schemas/schema/core-event-schema/unsigned_prop.yaml:30]: yea, might as well improve it while we're here. Have applied manually (line wrapping)

File: event-schemas/schema/core-event-schema/unsigned_prop.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+description: Contains optional extra information about the event.
+properties:
+  age:
+    description: The time in milliseconds that has elapsed since the event was
+      sent. This field is generated by the local homeserver, and may be incorrect
+      if the local time on at least one of the two servers is out of sync, which can
+      cause the age to either be negative or greater than it actually is.
+    type: integer
+  redacted_because:
+    description: Optional. The event that redacted this event, if any.
```
Comment (uhoreg) [inline event-schemas/schema/core-event-schema/unsigned_prop.yaml:26]: Also from the "let's fix this while we're in the area" department, it's kind of weird that this field says "Optional", while the other fields here (which are also optional) do not.  We should either make them all say "Optional" (incl. `invite_room_state` in `m.room.member`, or remove this one.

```suggestion
    description: The event that redacted this event, if any.
```

Comment (turt2live) [reply event-schemas/schema/core-event-schema/unsigned_prop.yaml:26]: yea... "Optional" followed by "This is basically required in some circumstances" is a bit confusing anyways.

**Pull request merged**